db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -87,6 +87,9 @@ bool DBImpl::ShouldRescheduleFlushRequestToRetainUDT(
   mutex_.AssertHeld();
   assert(flush_req.cfd_to_max_mem_id_to_persist.size() == 1);
   ColumnFamilyData* cfd = flush_req.cfd_to_max_mem_id_to_persist.begin()->first;
+  if (cfd->GetAndClearFlushSkipReschedule()) {
```

Comment:
This is more delicate than I initially expected, although aligned with the limitation mentioned in the unit test. Why not bluntly disable rescheduling on all flushes for the CF while a manual flush is happening? It could be disabled between the points where we discussed having callbacks (post-schedule/pre-wait, post-wait). Or maybe even start the disable pre-schedule.

Reply:
> Or maybe even start the disable pre-schedule.

Yeah it would need to be really early - before `WaitUntilFlushWouldNotStallWrites()`

Reply:
That's a very good question. The selected approach is trying to strike a balance between making sure as much as possible (should be 100%) that all manual flushes should proceed and not hang, while not incorrectly let unrelated automatic flushes slip through the cracks. Although the latter requirement is not that important. 

This involves communication between two threads, the thread that initiated the manual flush, and the background thread that does the check and flush the memtables. The blunt disabling approach only involves the thread that initiated the manual flush, my concern is there is a possibility that its intentions cannot reach the background thread.

It's a good point if we sacrifice the aformentioned balance a little bit and move the disabling reschedule early to before `WaitUntilFlushWouldNotStallWrites`, it can take care of that issue too.  I think that issue can be solved with an alternative approach. The root cause is the rescheduling checking logic has a different definition of what write stall is about to happen means, vs what `WaitUntilFlushWouldNotStallWrites` thinks.  It can be solved if we make them use the same criteria.

Reply:
> I think that issue can be solved with an alternative approach. The root cause is the rescheduling checking logic has a different definition of what write stall is about to happen means, vs what `WaitUntilFlushWouldNotStallWrites` thinks. It can be solved if we make them use the same criteria.

That works for me. Yeah I think the approach you have here is good.